test/lit/exec/fuzzing-api.wast

```diff
@@ -118,6 +120,139 @@
+  (call $log-i32
+   (call $call.ref.catch
+    (ref.null func)
+   )
+  )
+ )
+
+ (func $legal (param $x i32) (result i32)
+  ;; Helper for the function below. All types here are legal for JS.
+  (call $log-i32
+   (i32.const 12)
+  )
+  (i32.const 34)
+ )
+
+ ;; CHECK:      [fuzz-exec] calling ref.calling.legal
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 12]
+ (func $ref.calling.legal (export "ref.calling.legal")
+  ;; It is fine to call-ref a function with params and results. The params get
+  ;; default values, and the results are ignored. All we will see here is the
+  ;; logging from the function, "12".
+  (call $call.ref
+   (ref.func $legal)
+  )
+ )
+
+ (func $illegal (param $x i64)
+  ;; Helper for the function below. The param, an i64, causes a problem: when we
+  ;; call from JS we provide 0, but 0 traps when it tries to convert to BigInt.
+  (call $log-i32
+   (i32.const 56)
+  )
+ )
+
+ ;; CHECK:      [fuzz-exec] calling ref.calling.illegal
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 1]
+ (func $ref.calling.illegal (export "ref.calling.illegal")
+  ;; The i64 param causes an error here, so we will only log 1 as a trap.
+  (call $log-i32
+   (call $call.ref.catch
+    (ref.func $illegal)
+   )
+  )
+ )
+
+ (func $illegal-v128 (param $x v128)
+  ;; Helper for the function below.
+  (call $log-i32
+   (i32.const 56)
+  )
+ )
+
+ ;; CHECK:      [fuzz-exec] calling ref.calling.illegal-v128
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 1]
+ (func $ref.calling.illegal-v128 (export "ref.calling.illegal-v128")
+  ;; As above, we trap on the v128 param, and log 1.
+  (call $log-i32
+   (call $call.ref.catch
+    (ref.func $illegal-v128)
+   )
+  )
+ )
+
+ (func $illegal-result (result v128)
+  ;; Helper for the function below. The result is illegal for JS.
+  (call $log-i32
+   (i32.const 910)
+  )
+  (v128.const i32x4 1 2 3 4)
+ )
+
+ ;; CHECK:      [fuzz-exec] calling ref.calling.illegal-result
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 1]
+ (func $ref.calling.illegal-result (export "ref.calling.illegal-result")
+  ;; The v128 result causes an error here, so we will log 1 as a trap. The JS
+  ;; semantics determine that we do that check *before* the call, so the logging
+  ;; of 910 does not go through.
+  (call $log-i32
+   (call $call.ref.catch
+    (ref.func $illegal-result)
+   )
+  )
+ )
+
+ (func $legal-result (result i64)
+  ;; Helper for the function below.
+  (call $log-i32
+   (i32.const 910)
+  )
+  (i64.const 90)
+ )
+
+ ;; CHECK:      [fuzz-exec] calling ref.calling.legal-result
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 910]
+ ;; CHECK-NEXT: [LoggingExternalInterface logging 0]
+ ;; CHECK-NEXT: warning: no passes specified, not doing any work
+ (func $ref.calling.legal-result (export "ref.calling.legal-result")
+  ;; Unlike v128, i64 is legal in a result. The JS VM just returns a BigInt.
+  (call $log-i32
+   (call $call.ref.catch
+    (ref.func $legal-result)
+   )
+  )
+ )
```

Comment:
We should also add a test showing that a real trap from the function body halts execution.

Reply:
Good idea, done.